### PR TITLE
[DEV-3537] Enhance NAICS Advanced Search

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Add support for hierarchical NAICS codes to the advanced search - i.e., if `11` is received, `111110`, `111120` etc. are all searched.

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [X] Matview impact assessment completed (N/A)
5. [X] Frontend impact assessment completed (N/A)
6. [X] Data validation completed
7. [X] Appropriate Operations ticket(s) created (N/A)
8. [ ] Jira Ticket [DEV-3567](https://federal-spending-transparency.atlassian.net/browse/DEV-3567):
    - [x] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No change to frontend, no change to matviews, no ops tickets needed.
```

### CODE REVIEW

File: usaspending_api/awards/v2/filters/matview_filters.py [hunk 1]
```diff
@@ -254,9 +254,13 @@ def keyword_parse(keyword):
                 queryset = queryset.filter(cfda_number__in=in_query)
 
         elif key == "naics_codes":
-            in_query = [v for v in value]
-            if len(in_query) != 0:
-                queryset = queryset.filter(naics_code__in=in_query)
+            naics_filter = Q()
+            for v in value:
+                if len(v) == 6:
```
Comment (BrianZito) [inline usaspending_api/awards/v2/filters/matview_filters.py]: Is this second check needed? Doesn't `110011` start with `110011`?

File: usaspending_api/search/tests/test_spending_by_award.py [hunk 2]
```diff
@@ -293,3 +293,136 @@ def test_date_range_search_with_two_ranges(client, awards_over_different_date_ra
...
+        "subawards": False,
+        "fields": ["Award ID"],
+        "sort": "Award ID",
+        "limit": 50,
+        "page": 1,
+        "filters": {
+            "award_type_codes": ["A", "B", "C", "D"],
+            "naics_codes": ["11"]
+        },
+    }
```
Comment (BrianZito) [inline usaspending_api/search/tests/test_spending_by_award.py]: seems that the only thing that changes in here is the naics codes array. Could we have a function that just plops that in and returns the full JSON? The result would be much shorter tests that make it very clear what's being checked.

Comment (emily-brents) [reply usaspending_api/search/tests/test_spending_by_award.py]: sure

**Comment by tony-sappe** [comment]
Closing PR since this effort was abandoned for now. Leave the branch in Github for future reference until the new NAICS filter is implemented.

**Pull request closed without merging**